### PR TITLE
CCS harmonization

### PR DESCRIPTION
**Pull request by SucyLi**
## Fixes

Fixes #https://github.com/FCP-INDI/C-PAC/issues/1465 by @SucyLi 

## Description

Add ccs-options pipeline!

## Tests

For HNU sub-0025428:
Correlation of minimal preprocessed timeseries 0.937
Correlation of timeseries after nuisance regression 0.908

After rerunning CCS:
Correlation of minimal preprocessed timeseries 0.95329249
Correlation of timeseries after nuisance regression 0.92543674

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository. 
- [x] My commit messages follow best practices.
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I updated the changelog.
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: CPAC/resources/configs/pipeline_config_ccs-options.yml [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+anatomical_preproc: 
+
+  # Non-local means filtering via ANTs DenoiseImage
+  non_local_means_filtering: [On]
+
+  brain_extraction:
+
+    # using: ['3dSkullStrip', 'BET', 'UNet', 'niworkflows-ants', 'FreeSurfer-ABCD', 'FreeSurfer-BET-Tight', 'FreeSurfer-BET-Loose']
+    # this is a fork option
+    using: ['FreeSurfer-BET-Tight', 'FreeSurfer-BET-Loose']
```
Comment (SucyLi) [inline CPAC/resources/configs/pipeline_config_ccs-options.yml:77]: Currently, we'll generate 3 anatomical masks in ccs-options config: default FreeSurfer mask, FreeSurfer-BET tight mask and FreeSurfer-BET loose mask. Even though we generate 3 masks, it'll only use default FreeSurfer mask to do brain extraction if no manual intervention. CCS will generate 3 masks at the first stage and let users choose the best mask for following steps. I am wondering if there's a better way to replicate CCS. Do we need to fork 3 pipelines with 3 anatomical masks?

File: CPAC/pipeline/cpac_pipeline.py [hunk 2]
```diff
@@ -744,13 +751,20 @@ def build_anat_preproc_stack(rpool, cfg, pipeline_blocks=None):
                     acpc_align_brain_with_mask
                     # outputs space-T1w_desc-brain_mask for later - keep the mask (the user provided)
                 ]
+                if 'FreeSurfer-BET-Tight' in cfg.anatomical_preproc[
+                    'anatomical_preproc']['brain_extraction']['using'] or 'FreeSurfer-BET-Loose' \
+                        in cfg.anatomical_preproc['anatomical_preproc']['brain_extraction']['using']:
```
Comment (sgiavasis) [inline CPAC/pipeline/cpac_pipeline.py:756]: Isn't this switch automatically handled by the node block/connector?

File: CPAC/resources/configs/pipeline_config_ccs-options.yml (hunk 1 above)
Comment (sgiavasis) [reply CPAC/resources/configs/pipeline_config_ccs-options.yml:77]: Like we've decided in the past for other "manual checkpoint" use cases - we can offer both.

- Users can simply enable only the mask generation and run the pipeline, then pick which mask they want and run the rest of the pipeline.
- Or, can fork all 3 and let CPAC blast through all options.

File: CPAC/func_preproc/func_preproc.py [hunk 3]
```diff
@@ -1662,6 +1677,149 @@ def bold_mask_anatomical_based(wf, cfg, strat_pool, pipe_num, opt=None):
...
+def bold_mask_ccs(wf, cfg, strat_pool, pipe_num, opt=None):
+    '''Generate the BOLD mask by basing it off of the anatomical brain.
+    Adapted from the BOLD mask method from the CCS pipeline.
+        https://github.com/TingsterX/CCS/blob/master/ccs_01_funcpreproc.sh#L89-L110
+    Node Block:
+    {"name": "bold_mask_ccs",
+     "config": ["functional_preproc"],
+     "switch": ["run"],
+     "option_key": ["func_masking", "using"],
+     "option_val": "CCS",
```
[the remark this answers is not in the document]
Comment (SucyLi) [reply CPAC/func_preproc/func_preproc.py]: We decide to call it `CCS_Anatomical_Refined`

File: CPAC/pipeline/cpac_pipeline.py (hunk 2 above)
Comment (SucyLi) [reply CPAC/pipeline/cpac_pipeline.py]: You're right! Resolved

**Pull request merged**